pubsubx/kgox/consumer.go

```diff
@@ -175,8 +183,9 @@ func (c *consumer) start(ctx context.Context) {
 			wrappedHandler := func(ctx context.Context, msgs []*messagex.Message) (outErrs []error, outErr error) {
 				defer func() {
 					if r := recover(); r != nil {
-						l.Errorf("panic while handling messages: %v", r)
 						outErr = errorx.InternalErrorf("panic while handling messages")
+						stackTrace := tracexx.GetStackTrace()
+						l.WithFields(logrusx.NewLogFields(semconv.ExceptionStacktrace(stackTrace))).Errorln("panic while handling messages")
```

Comment:
We need pass the context here to log trace and span info
```suggestion
						l.WithContext(ctx).WithFields(logrusx.NewLogFields(semconv.ExceptionStacktrace(stackTrace))).Errorln("panic while handling messages")
```

Reply:
Is there a reason where are not using the WithStack() method ?

Reply:
Yeah good call. Won't pass it for the `RecoverWithStackTrace` as we can pass the configured logger with it.